sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
This is not required. You just need put [this if block](https://github.com/sympy/sympy/pull/12224/files#diff-eec0422923e8f100745c015cd8fdd6cfR861) above Line 848.